crates/revm/src/evm_impl.rs

```diff
@@ -409,7 +411,26 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> Transact<DB::Error>
                     Output::Create(return_value, _) => return_value,
                 },
             },
-            SuccessOrHalt::Halt(reason) => ExecutionResult::Halt { reason, gas_used },
+            SuccessOrHalt::Halt(reason) => {
+                // Post-regolith, if the transaction is a deposit transaction and the
+                // output is a contract creation, increment the account nonce even if
+                // the transaction halts.
+                #[cfg(feature = "optimism")]
+                {
+                    let is_creation = matches!(output, Output::Create(_, _));
+                    let regolith_enabled = RegolithSpec::enabled(self.data.env.cfg.spec_id);
```

Comment:
This looks incorrect - should be using `GSPEC::enabled(REGOLITH)` here I believe.

Reply:
Ah shoot, great catch